apps/src/codebridge/FilePreview/styles/html-preview-header.module.scss

```diff
@@ -1,106 +1,68 @@
-$url-bar-height: 24px;
-$vertical-padding-offset: 4px;
-$url-bar-content-height: calc($url-bar-height - $vertical-padding-offset);
-
 .previewHeaderContainer {
   background-color: var(--background-neutral-tertiary);
   border-bottom: 1px solid var(--borders-neutral-primary);
-  padding: 8px 12px;
+  padding: 0.5rem;
   display: flex;
   align-items: center;
-  gap: 8px;
+  gap: 0.5rem;
   min-width: 0; // Allow container to shrink
-  
+
   .urlBarContent {
     display: flex;
     flex: 1;
-    min-width: 0; // Allow URL bar to shrink
+    min-width: 0.5rem; // Allow URL bar to shrink
     border: 1px solid var(--borders-neutral-strong);
     border-radius: var(--border-radius-br-s, 4px);
     background-color: var(--background-neutral-primary);
-    height: $url-bar-height;
     overflow: visible;
-    padding: 2px 2px; // Inner padding to accommodate focus outlines
     box-sizing: border-box;
     align-items: center;
+    container: urlBarContainer / inline-size;
   }
 
-  // Wrapper for the URL buttons and refresh button
-  .urlButtonsWrapper {
-    display: flex;
-    background-color: var(--background-neutral-primary);
-    height: $url-bar-content-height;
-    overflow: hidden;
-    box-sizing: border-box;
-    align-items: center;
-  }
-  
-  // Add spacing between nav buttons and input
-  .navButtonsWrapper {
-    margin-right: 2px;
+  // Hide icon buttons when the URL bar is too small
+  @container urlBarContainer (width < 100px) {
+    .iconButton {
+      display: none;
+    }
   }
-  
-  // Add spacing between input and refresh button
-  .refreshButtonWrapper {
-    margin-left: 2px;
+
+  .iconButton {
+    padding: 0.2188rem;
   }
 
-  .urlButton {
-    color: var(--text-neutral-quaternary);
-    margin: 0;
-    
-    // Override button sizing for URL bar to fit within URL bar content container
-    padding: 0.125rem; // Smaller padding to fit within height constraints
-    min-width: auto; // Remove min-width constraint
-    width: $url-bar-content-height;
-    height: $url-bar-content-height;
-    
-    // Override button focus styles for tight URL bar layout
-    &:focus-visible {
-      outline: 1px solid var(--borders-brand-teal-primary);
-      outline-offset: -1px; // Move outline slightly inside button border so that it doesn't get cut off by adjacent elements.
-      border-radius: 0.25rem;
-      position: relative;
+  .navButtonsWrapper {
+    z-index: 1;
+    display: flex;
+    flex-direction: row;
+
+    // Reverse button order for RTL languages
+    html[dir='rtl'] & {
```

Comment:
Just curious - would we reverse the order of url bar and buttons as well for rtl?

Reply:
Not sure what you mean, can you clarify? The bar and buttons are already flipping due to the flexbox wrapper that contains them so this already looks correct for rtl.

Reply:
oh gotcha - nice!